### PR TITLE
Implement std Error for client::Error

### PR DESCRIPTION
**Pull request by CodeSandwich**
Fixes https://github.com/radicle-dev/radicle-registry/issues/226

There are a few changes in the API to make it more conventional. @xla please confirm if they are OK with you.

### CODE REVIEW

**Comment by CodeSandwich** [comment]
> Is there a technical reason to drop the period marks at the end of the affected sentences?

No, it just feels right when single sentence lists don't have dots.

File: client/src/error.rs [hunk 1]
```diff
@@ -13,22 +13,40 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+use failure::{Compat, Fail};
 use jsonrpc_core_client::RpcError;
 use parity_scale_codec::Error as CodecError;
+use thiserror::Error;
```
Comment (NunoAlexandre) [inline client/src/error.rs:19]: following to the discussion around ambiguity, I'd suggest us applying that here already:

```suggestion
use thiserror::Error as ThisError;
```

File: client/src/error.rs [hunk 2]
```diff
@@ -13,22 +13,40 @@
...
 
-/// Error that may be returned by any of the [crate::ClientT] methods.
-#[derive(Debug, derive_more::From, derive_more::Display, derive_more::TryInto)]
+/// Error that may be returned by any of the [crate::ClientT] methods
+#[derive(Debug, Error)]
```
Comment (NunoAlexandre) [inline client/src/error.rs:22]:
```suggestion
#[derive(Debug, ThisError)]
```

File: client/src/error.rs (hunk 1 above)
Comment (CodeSandwich) [reply client/src/error.rs]: Interesting case of ambiguity, unclear for humans, but easy for the compiler, good catch! I'd rather avoid aliasing if not necessary.

Fixed, but differently.

File: client/src/error.rs (hunk 2 above)
Comment (CodeSandwich) [reply client/src/error.rs]: Fixed, but differently

File: client/src/error.rs (hunk 1 above)
Comment (NunoAlexandre) [reply client/src/error.rs]: ahah indeed, it;s usually the way around. Your fix is good :+1:

**Pull request merged**